core/lib/multivm/src/versions/vm_latest/bootloader/state.rs

```diff
@@ -196,13 +211,24 @@ impl BootloaderState {
                     self.execution_mode,
                     num == 0,
                     self.subversion,
+                    num == 0,
+                    self.get_preexisting_interop_roots_number(),
+                    self.get_preexisting_blocks_number(),
```

Comment:
this function became ultra complex with multiple bools, could we specify a few structs for it? for making it more obvious